### PR TITLE
Revise README and add doc for El attention

### PR DESCRIPTION
**Pull request by yuyan2do**

### CODE REVIEW

File: examples/EL-attention/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+| `transformer` | 270M | SQuAD 1.1 |  436.4 (190.3)   | 699.7 (260.3)   | **1.6x** (1.4x)  |
+| `bart.large` | 400M  | XSum | 42.6 (15.0) | 107.8 (44.9)  | **2.5x** (3.0x)  |
+| `bart.large` | 400M | CNN/DailyMail | 13.0 (7.0) | 40.0 (19.5) | **3.1x** (2.8x) |
+| `gpt-2.small` | 117M  | CNN/DailyMail | 14.7 (9.0) | 26.2 (13.6) | **1.8x** (1.5x) |
+| `gpt-2.medium` | 345M  | CNN/DailyMail | 5.9 (3.6) | 10.4 (5.9) | **1.8x** (1.6x) |
+
+## Example usage
+
+##### Command line:
+
```
Comment (NickNickGo) [inline examples/EL-attention/README.md:76]: Add usage here.

File: examples/EL-attention/summarize.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    bart = bart.eval()
+    if torch.cuda.is_available():
+        bart = bart.cuda().half()
+
+    # TODO make this step automatic
+    if args.use_el_attn:
+        #for model in bart.models:
+        bart.model.transpose_enc_dec_kv_proj()
+        #bart.model.set_beam_size(args.bsz)
+        #param = dict({'beamable_mm_beam_size': args.bsz})
```
Comment (NickNickGo) [inline examples/EL-attention/summarize.py:109]: Remove comments and extra blank lines. Remove #TODO is already done.

**Review by NickNickGo — APPROVED** [review]
Thanks Yu for the PR ! Looks Good to me. 

Minor comments: 
1. Can we add brief description of notebook in first cell? 
2. We can put "Inference with MultiHead-Attention" above "Inference with EL-Attention", and add small text cell summarizing gain over multihead. This could make notebook more readable for user.
3.  Multi head attention cell output is not printed.
4. Last cell - Evaluate output has some errors.

File: README.md [hunk 3]
```diff
@@ -4,6 +4,11 @@
 
 FastSeq provides efficient implementation of popular sequence models (e.g. [Bart](https://arxiv.org/pdf/1910.13461.pdf), [ProphetNet](https://github.com/microsoft/ProphetNet)) for text generation, summarization, translation tasks etc. It automatically optimizes inference speed based on pupular NLP toolkits (e.g. [FairSeq](https://github.com/pytorch/fairseq) and [HuggingFace-Transformers](https://github.com/huggingface/transformers)) without accuracy loss. All these can be easily done (no need to change any code/model/data if using our command line tool, or simply add one-line code `import fastseq` if using source code).
 
+## Features:
+* EL-Attention: Memory Efficient Lossless Attention for Generation ([Yan et al., 2021](https://arxiv.org/abs/2105.04779))
+* GPU-based Block N-Gram Repeats
+* Asynchronous Pipeline for Postprocess
+
```
Comment (feihugis) [inline README.md:11]: Add the optimized cache in the attention?

File: examples/EL-attention/EL_attention_Demo.ipynb [hunk 4]
```diff
@@ -0,0 +1,694 @@
...
+      "execution_count": 13,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "Error: Could not find or load main class edu.stanford.nlp.process.PTBTokenizer\n",
+            "Caused by: java.lang.ClassNotFoundException: edu.stanford.nlp.process.PTBTokenizer\n",
+            "Error: Could not find or load main class edu.stanford.nlp.process.PTBTokenizer\n",
+            "Caused by: java.lang.ClassNotFoundException: edu.stanford.nlp.process.PTBTokenizer\n",
+            "/bin/bash: files2rouge: command not found\n"
```
Comment (feihugis) [inline examples/EL-attention/EL_attention_Demo.ipynb:676]: Is it expected to show error messages here?

File: examples/EL-attention/summarize.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+XSUM_KWARGS = dict(beam=6, lenpen=1.0, max_len_b=60, min_len=10, no_repeat_ngram_size=3)
+CNN_KWARGS = dict(beam=4, lenpen=2.0, max_len_b=140, min_len=55, no_repeat_ngram_size=3)
+
+
+@torch.no_grad()
+def generate(bart, infile, outfile="bart_hypo.txt", bsz=32, n_obs=None, **eval_kwargs):
+    count = 1
+
+    # if n_obs is not None: bsz = min(bsz, n_obs)
```
Comment (feihugis) [inline examples/EL-attention/summarize.py:15]: remove this line?

File: README.md (hunk 3 above)
Comment (yuyan2do) [reply README.md:11]: EL-Attention covered all of our attention opt.

File: examples/EL-attention/EL_attention_Demo.ipynb (hunk 4 above)
Comment (yuyan2do) [reply examples/EL-attention/EL_attention_Demo.ipynb:676]: This part cannot run. I will change this cell from code to text.

File: examples/EL-attention/summarize.py (hunk 5 above)
Comment (yuyan2do) [reply examples/EL-attention/summarize.py:15]: Will remove

File: examples/EL-attention/summarize.py (hunk 2 above)
Comment (yuyan2do) [reply examples/EL-attention/summarize.py:109]: removed comments. TODO is not done, keep it here.

**Comment by JiushengChen** [comment]
Merge new changes in master branch.

**Comment by yuyan2do** [comment]
> Merge new changes in master branch.

Merged. Please help review.

**Review by JiushengChen — APPROVED** [review]

**Review by feihugis — APPROVED** [review]

**Pull request merged**